app/api/threads/list_threads.feature

```diff
@@ -0,0 +1,172 @@
+      | Charlotte           | EtienneKleinHasValidated5             | Université           | closed                  | 2021-12-20T00:00:00Z | EtienneKlein is_mine=0 -> List thread ok: part of helper group, closed thread for less than 2 weeks and validated item |
+      | Charlotte           | EtienneKleinCanWatchAnswer4           | Université           | closed                  | 2021-12-20T00:00:00Z | EtienneKlein is_mine=0 -> List thread ok: can_watch >= answer                                                          |
+      | Charlotte           | EtienneKleinHasValidated6             | Université           | closed                  | 2021-11-00T00:00:00Z | EtienneKlein is_mine=0 -> List thread notok: closed for more than 2 weeks                                              |
+      | Charlotte           | EtienneKleinCanWatchAnswer5           | Université           | closed                  | 2021-11-00T00:00:00Z | EtienneKlein is_mine=0 -> List thread ok: can_watch >= answer                                                          |
+      | Baptiste            | BaptisteCanViewInfo                   |                      |                         |                      | Baptiste is_mine=1 -> notok: can_view < content                                                                        |
+      | Baptiste            | BaptisteCanViewContent1               |                      |                         |                      | Baptiste is_mine=1 -> ok: can_view >= content                                                                          |
+      | Thibaut             | BaptisteCanViewContent2               |                      |                         |                      | Baptiste is_mine=1 -> notok: not the participant                                                                       |
+      | Baptiste            | BaptisteCanViewContentWithDescendants |                      |                         |                      | Baptiste is_mine=1 -> ok: can_view >= content                                                                          |
+      | DavidBowie          |                                       |                      |                         |                      |                                                                                                                        |
+    And the time now is "2022-01-01T00:00:00Z"
+
+  Scenario: Should have all the fields properly set, including first_name and last_name when the access is approved
+    Given I am MarieCurie
+    And there is a group Laboratory referenced by @Laboratory
+    And I am a manager of the group Laboratory
+    And I can watch the group Laboratory
+    And there are the following users:
+      | @reference      | login          | first_name | last_name |
+      | @AlbertEinstein | AlbertEinstein | Albert     | Einstein  |
+      | @PaulDirac      | PaulDirac      | Paul       | Dirac     |
```

Comment:
I don't get why I have "@..." while I have no "@..." for user definitions at the top of the file.
I like the gherkin convention that "background" rules are the same as "scenario" ones.

Reply:
The implementation of those @... feels a bit hacky. There're only here to accommodate for when we need to explicitly get the ID of something, in those cases:

url:
- `When I send a GET request to "/threads?watched_group_id=@Laboratory"`

responses:
- `And the response at $[0].participant.id should be "@Mary"`
-
```
"participant": {
            "id": "@AlbertEinstein",
            "login": "AlbertEinstein",
            "first_name": "Albert",
            "last_name": "Einstein"
},
```


**Internally, the references already exist** because they are used by the Gherkin features. For example:
`And I am a manager of the group Laboratory and can watch its members`
Laboratory is a reference here, but we know with the context that it's a reference. We could add @... everywhere but it would not be natural.

In the urls and in the responses, we don't know when we're referring to an entity's id.

This means that statements like:
`And the group University is referenced by @University`
could be removed, and the test would still work. But this line doesn't do nothing: it's possible to replace the name of the reference and it would create this other reference.

The options that we have here are:
- Replacing every `Name` by `@Name`, in all tables and all Gherkin definitions, every time we refer to an entity. It would remove ambiguity but reduce readability.
- Removing all the definitions of `@references` in tables and `referenced by` in features because it works without it. Simpler but has a feeling of "magic" in it.
- A hybrid version like we have now, with explicit references declarations, even if it's technically not necessary. Not sure if any better than the two other options?
- Another solution?